### PR TITLE
linter: Introduce a stricter formatter `gofumpt`

### PR DESCRIPTION
**Pull request by EdDev**
### What this PR does

A stricter formater allows to reduce formatting and style opinionated comments in reviews.
It provides consistency and helps with code readability.

[gofumpt](https://github.com/mvdan/gofumpt) seems to be popular among the Go community, therefore chosen for this task.

Used the `extra` flag to add all rules.

To automatically format the code, use: `make gofumpt` or:
```
gofumpt -l -w -extra \
  pkg/instancetype \
  pkg/libvmi \
  pkg/network/admitter \
  pkg/network/namescheme \
  pkg/network/domainspec \
  pkg/network/deviceinfo \
  tests/console \
  tests/libnet \
  tests/libnode \
  tests/libconfigmap \
  tests/libpod \
  tests/libvmifact \
  tests/libsecret
```

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by EdDev** [comment]
There has been a change in the builder Dockerfile [1] and therefore a new image needs to be created for this change.

@brianmcarey , I am rebasing this PR in order to include the additions.
Could you please help in creating a new builder image? (the prev can be removed from the registry)

[1] https://github.com/kubevirt/kubevirt/commit/13889fb0b57b115e7b9b91ad73b39085dc0e06cd

/hold

**Comment by EdDev** [comment]
change: Rebase

**Comment by EdDev** [comment]
change: Updated the builder image.

**Comment by EdDev** [comment]
> There has been a change in the builder Dockerfile [1] and therefore a new image needs to be created for this change.
> 
> @brianmcarey , I am rebasing this PR in order to include the additions. Could you please help in creating a new builder image? (the prev can be removed from the registry)
> 
> [1] [13889fb](https://github.com/kubevirt/kubevirt/commit/13889fb0b57b115e7b9b91ad73b39085dc0e06cd)
> 
> /hold

Done
/unhold

**Review by enp0s3 — COMMENTED** [review]
Hey! Can you please update the PR description section "Special notes for your reviewer"
IIUC the comment there is redundant

**Comment by EdDev** [comment]
> Hey! Can you please update the PR description section "Special notes for your reviewer" IIUC the comment there is redundant

Done

File: hack/builder/Dockerfile [hunk 1]
```diff
@@ -113,6 +113,7 @@ RUN set -x && \
 
 RUN set -x && \
     source /etc/profile.d/gimme.sh && \
+    go install -v mvdan.cc/gofumpt@v0.6.0 && \
```
Comment (orelmisan) [inline hack/builder/Dockerfile:116]: AFAIU, this tool is a [part](https://golangci-lint.run/usage/linters/#gofumpt) of `golangci-lint`.
Will it be installed twice after this change?

**Review by orelmisan — COMMENTED** [review]
Thank you @EdDev 
I personally like most of the formatting this tool does.

File: hack/builder/Dockerfile (hunk 1 above)
Comment (EdDev) [reply hack/builder/Dockerfile:116]: Per my understanding, it is embedded into the `golangci-lint` but not performing the auto-formatting.
It is installed here in order to allow the auto-formatting, not related to running the linter validation.

**Review by orelmisan — APPROVED** [review]
Thank you

**Comment by enp0s3** [comment]
/approve
/hold
@EdDev Thanks for the PR, sounds good to me. Feel free to unhold upon rebase

**Review by orelmisan — APPROVED** [review]

**Comment by EdDev** [comment]
> /approve /hold @EdDev Thanks for the PR, sounds good to me. Feel free to unhold upon rebase

Rebased.

/unhold

**Pull request merged**